lhotse/recipes/ami.py

```diff
@@ -101,179 +238,277 @@ class AmiSegmentAnnotation(NamedTuple):
+
+def prepare_supervision_other(
+    audio: RecordingSet,
+    annotations: Dict[str, List[AmiSegmentAnnotation]]
+) -> SupervisionSet:
+    annotation_by_id = {
+        (key[0]): annot
+        for key, annot in  annotations.items()
+    }
 
-            # Make the begin/end points list
-            seg_btimes = [starttime] + puncts_times
-            seg_btimes.pop()
-            seg_times = [AmiSegmentAnnotation(
-                text=t,
-                speaker=speaker,
-                gender=speaker[0],
-                begin_time=b,
-                end_time=e
-            ) for t, b, e in zip(trans, seg_btimes, puncts_times)]
-
-            wav_name = f'{meet_id}.Headset-{int(channel)}.wav'
-            if wav_name not in anotations:
-                anotations[wav_name] = []
-            anotations[wav_name].append(seg_times)
-
-    return anotations
-
+    segments = []
+    for recording in audio:
+        annotation = annotation_by_id.get(recording.id)
+        # In these mic settings, all sources (1 for ihm-mix and sdm and 16 for mdm)
+        # will share supervision.
+        source = recording.sources[0]
+        if annotation is None:
+            logging.warning(f'No annotation found for recording {recording.id}')
+            continue
+        
+        if (len(source.channels) > 1):
+            logging.warning(f'More than 1 channels in recording {recording.id}. '
+                            f'Creating supervision for channel 0 only.')
+
+        for seg_idx, seg_info in enumerate(annotation):
+            duration = seg_info.end_time - seg_info.begin_time
+            if duration > 0:
+                segments.append(SupervisionSegment(
+                    id=f'{recording.id}-{seg_idx}',
+                    recording_id=recording.id,
+                    start=seg_info.begin_time,
+                    duration=duration,
+                    channel=0,
+                    language='English',
+                    speaker=seg_info.speaker,
+                    gender=seg_info.gender,
+                    text=seg_info.text
+                ))
+    return SupervisionSet.from_segments(segments)
 
 def prepare_ami(
         data_dir: Pathlike,
         output_dir: Optional[Pathlike] = None,
+        mic: Optional[str] = 'ihm',
+        partition: Optional[str] = 'full-corpus',
+        max_pause: Optional[float] = 0.0
 ) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
     """
     Returns the manifests which consist of the Recordings and Supervisions
-
     :param data_dir: Pathlike, the path of the data dir.
     :param output_dir: Pathlike, the path where to write the manifests.
-    :return: a Dict whose key is ('train', 'dev', 'eval'), and the value is Dicts with keys 'audio' and 'supervisions'.
+    :param mic: str {'ihm','ihm-mix','sdm','mdm'}, type of mic to use.
+    :param partition: str {'full-corpus','full-corpus-asr','scenario-only'}, AMI official data split
```

Comment:
Can you add the same comment you made in the PR about which partition to select for ASR and which for diarization?